### PR TITLE
Make INotifyIdle, INotifyBecomingIdle & ITick require explicit implementation

### PR DESCRIPTION
**Pull request by reaperrr**
We should finish what we started at some point, and to do that we need to actually bother making some headway on this matter.

This PR finishes the job on the two `Idle` interfaces and the very common `ITick` interface.

Reviewing commit by commit is recommended, most of the stuff is simple, but going commit by commit makes the diff more readable.

### CODE REVIEW

**Review by pchote — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs [hunk 1]
```diff
@@ -122,7 +122,12 @@ void INotifyAttack.PreparingAttack(Actor self, Target target, Armament a, Barrel
 
 		void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barrel) { }
 
-		public virtual void Tick(Actor self)
+		void ITick.Tick(Actor self)
+		{
+			Tick(self);
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs]: This seems to add more than the explicit interface?

Comment (reaperrr) [reply OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs]: This is the only way to make ITick explicit without sacrificing the overridability.

**Comment by reaperrr** [comment]
Rebased.

File: OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs (hunk 1 above)
Comment (reaperrr) [reply OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs]: Note that we already use the same approach in various places (though more often for `INotifyCreated.Created`), so this is both consistent with other traits and proven & accepted as solutions for these cases.

**Review by Mailaender — APPROVED** [review]
Okay. Let's get over with this. Sorry for everyone who has to rebase.

**Pull request merged**